plugins/catalog-backend/src/search/DefaultCatalogCollatorFactory.ts

```diff
@@ -23,30 +24,32 @@ import {
   CatalogClient,
   GetEntitiesRequest,
 } from '@backstage/catalog-client';
-import { stringifyEntityRef } from '@backstage/catalog-model';
-import { Config } from '@backstage/config';
 import { DocumentCollatorFactory } from '@backstage/plugin-search-common';
 import {
   catalogEntityReadPermission,
   CatalogEntityDocument,
 } from '@backstage/plugin-catalog-common';
 import { Permission } from '@backstage/plugin-permission-common';
 import { Readable } from 'stream';
-import { getDocumentText } from './util';
+import { CatalogCollatorEntityTransformer } from './CatalogCollatorEntityTransformer';
+import { defaultCatalogCollatorEntityTransformer } from './defaultCatalogCollatorEntityTransformer';
+import { stringifyEntityRef } from '@backstage/catalog-model';
 
 /** @public */
 export type DefaultCatalogCollatorFactoryOptions = {
   discovery: PluginEndpointDiscovery;
   tokenManager: TokenManager;
+  type?: string;
```

Comment:
I'm not sure this is a means of customization that we want to support.  What's the use-case?

Reply:
Could you clarify what the purpose of adding this ability is? If it is for search filtering purposes, I recommend keeping the index type as "software-catalog" and adding the type in your transformed entity documents.

Reply:
removed ability to change type